themes/hawaii50/hawaii50.theme.bash

```diff
@@ -170,7 +174,7 @@ function limited_pwd() {
 	# Replace $HOME with ~ if possible
 	RELATIVE_PWD=${PWD/#$HOME/\~}
 
-	local offset=$((${#RELATIVE_PWD} - $MAX_PWD_LENGTH))
+	local offset=$((${#RELATIVE_PWD} - MAX_PWD_LENGTH))
 
 	if [ $offset -gt "0" ]; then
```

Comment:
I'm not sure why the shellcheck doesn't complain about this unquoted `$offset`. Maybe it is because `IFS` is not modified inside this file and `offset` is ensured to contain a number in this context? However, the shell configuration should anticipate an arbitrary `IFS` value set outside the file by the user.

Reply:
I've noticed that recent ShellCheck releases have become better at knowing when variables have a value that can only be a number. But I disagree that it should anticipate an arbitrary `IFS` value - the value deduction is just too handy and it covers 99% of setups.

Reply:
> But I disagree that it should anticipate an arbitrary `IFS` value - the value deduction is just too handy and it covers 99% of setups.

I wouldn't argue that there would be a sane setup with `IFS` including numbers, but it is a valid usage to temporarily set `IFS` to a number, do something, and then revert it.

```console
$ oldifs=${IFS-$' \t\n'}
$ IFS='blahblah'
$ do-something
$ IFS=$oldifs
```

In such a case, `[ $offset -gt "0" ]` may be broken temporarily for the third and fourth prompts.

Actually, one could simply rewrite it as `((offset > 0))`, which is more obvious and easier to understand.